ambry-tools/src/main/java/com.github.ambry/store/DumpData.java

```diff
@@ -667,20 +651,39 @@ private void compareIndexEntriesToLogContent(File indexFile, File replicaDirecto
               blobValue.getSize() + " Original Message Offset " + blobValue.getOriginalMessageOffset() +
               " Flag " + blobValue.getFlags() + "\n";
           boolean isDeleted = blobValue.isFlagSet(IndexValue.Flags.Delete_Index);
-          if (deletedMsgInfo != null && isDeleted) {
-            deletedMsgInfo.put(blobValue.getOriginalMessageOffset(), blobValue.getOffset());
-          }
           boolean success =
               dumpDataHelper.readFromLogAndVerify(randomAccessFile, blobValue.getOffset(), key.getID(), blobValue,
-                  coveredRangesInLog);
+                  coveredRanges);
           if (!success) {
             logger.error("Failed for Index Entry " + msg);
           }
+          if (isDeleted) {
+            long originalOffset = blobValue.getOriginalMessageOffset();
+            deleteMsgsOriginalOffsets.add(originalOffset);
+            if (originalOffset != -1 && !coveredRanges.containsKey(originalOffset) &&
+                originalOffset >= startOffset) {
+              randomAccessFile.seek(originalOffset);
+              DumpDataHelper.BlobRecordInfo blobRecordInfo =
+                  dumpDataHelper.readSingleRecordFromLog(randomAccessFile, originalOffset);
+              coveredRanges.put(originalOffset, originalOffset + blobRecordInfo.totalRecordSize);
+              logger.trace(
+                  "PUT Record {} added with orig offset {} and end offset {} for a delete msg " + "{} at offset {} ",
+                  blobRecordInfo.blobId, originalOffset, (originalOffset + blobRecordInfo.totalRecordSize), key.getID(),
+                  blobValue.getOffset());
+              if (!blobRecordInfo.blobId.getID().equals(key.getID())) {
+                logger.error("BlobId value mismatch between delete record " + key.getID() + " and put record "
+                    + blobRecordInfo.blobId.getID());
+              }
+            } else {
+              if (!coveredRanges.containsKey(originalOffset) && startOffset > originalOffset) {
+                logger.trace("Put Record ignored at {} with delete msg's offset {} prior to startOffset {} ",
```

Comment:
log message is a bit misleading. Seems to say that the deleted message's offset is prior to start offset (which it is not). 

Should be something like "Put Record at {} with delete msg offset {} ignored because it is prior to startOffset {}"